### PR TITLE
Store message_key on validation exception

### PR DESCRIPTION
**Pull request by stjhimy**
Also add missing spec for validation exception.

### CODE REVIEW

**Comment by stjhimy** [comment]
huh, it might be duplicated from #1210 @dblock 
should I remove the message_key commit?

**Comment by dblock** [comment]
Yes but I am glad this one has tests. Can you please squash the commits?

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 * [#1216](https://github.com/ruby-grape/grape/pull/1142): Fix JSON error response when calling `error!` with non-Strings - [@jrforrest](https://github.com/jrforrest).
 * [#1225](https://github.com/ruby-grape/grape/pull/1225): Fix `given` with nested params not returning correct declared params - [@JanStevens](https://github.com/JanStevens).
+* [#1227](https://github.com/ruby-grape/grape/pull/1227): Store message_key on validation exception - [@stjhimy](https://github.com/sthimy).
```
Comment (dblock) [inline CHANGELOG.md]: Quote `message_key` and maybe make the class specific, so Grape::Exceptions::Validation? NBD

**Comment by stjhimy** [comment]
@dblock done.

**Comment by dblock** [comment]
Merged via https://github.com/ruby-grape/grape/commit/c54cd8f0487480e7633abb92ae03b138c080e37e.

**Pull request closed without merging**